modules/460-log-shipper/hooks/internal/vector/destination/vector.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2021 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package destination
+
+import (
+	"github.com/deckhouse/deckhouse/modules/460-log-shipper/hooks/internal/impl"
+	"github.com/deckhouse/deckhouse/modules/460-log-shipper/hooks/internal/v1alpha1"
+)
+
+type Vector struct {
+	CommonSettings
+
+	Version string `json:"version,omitempty"`
+
+	Address string `json:"address"`
+
+	TLS VectorTLS `json:"tls,omitempty"`
+
+	Keepalive VectorKeepalive `json:"keepalive,omitempty"`
+}
+
+type VectorTLS struct {
+	CommonTLS         `json:",inline"`
+	VerifyCertificate bool `json:"verify_certificate"`
+	Enabled           bool `json:"enabled"`
+}
+
+type VectorKeepalive struct {
+	TimeSecs int `json:"time_secs"`
+}
+
+func NewVector(name string, cspec v1alpha1.ClusterLogDestinationSpec) impl.LogDestination {
+	spec := cspec.Vector
+
+	// Disable buffer. It is buggy. Vector developers know about problems with buffer.
```

Comment:
Is buffer still broken? As I remember there was some fix for it

Reply:
We have a different issue for buffer problems. It is still no use for our needs.

Reply:
https://github.com/deckhouse/deckhouse/issues/1352